ql/src/java/org/apache/hadoop/hive/ql/exec/repl/ReplDumpTask.java

```diff
@@ -424,11 +518,13 @@ private boolean validDump(Path dumpDir) throws IOException {
     return false;
   }
 
-  private boolean shouldDump(Path previousDumpPath) throws IOException {
+  private boolean shouldDump(Path previousDumpPath, boolean isPrevDumpFailoverReady) throws IOException {
     //If no previous dump means bootstrap. So return true as there was no
     //previous dump to load
     if (previousDumpPath == null) {
       return true;
+    } else if (isPrevDumpFailoverReady) {
+      return false;
```

Comment:
What if in the previous run right before dump ack creation it failed. The failover ready marker will still be there and it must go for another dump no?

Reply:
Yes, in that case, it won't be treated as valid dump and execution would go to getCurrentDumpPath to resume the previous failed dump.